packages/graphql/src/translate/create-relationship-validation-str.ts

```diff
@@ -0,0 +1,48 @@
+import { Node } from "../classes";
+import { RELATIONSHIP_REQUIREMENT_PREFIX } from "../constants";
+import { Context } from "../types";
+
+function createRelationshipValidationStr({
+    node,
+    context,
+    varName,
+}: {
+    node: Node;
+    context: Context;
+    varName: string;
+}): string {
+    let relationshipValidationStr = "";
+
+    const nonNullRelationFields = node.relationFields.filter((field) => {
+        const isRequired = field.typeMeta.required;
+        const isArrayTypeRequired = field.typeMeta.arrayTypeRequired;
+        const isArray = field.typeMeta.array;
+        const isNotUnionOrInterface = Boolean(field.union) || Boolean(field.interface);
+
+        if (isArray) {
+            return isArrayTypeRequired && !isNotUnionOrInterface;
+        }
+
+        return isRequired && !isNotUnionOrInterface;
+    });
```

Comment:
Consider moving this to a separate function: 
```suggestion
    const nonNullRelationFields = filterNonNullFields(node.relationFields);
```
(This is a suggestion, I'm happy to keep it as is too)

Reply:
I'm going to keep this as it is. I feel the function is pretty small in size and breaking it down will just add unnecessary context switching for future readers.